app/src/main/java/fr/free/nrw/commons/CommonsApplication.java

```diff
@@ -275,7 +280,6 @@ public void clearApplicationData(Context context, LogoutListener logoutListener)
                     //TODO: fix preference manager
                     defaultPrefs.clearAll();
                     defaultPrefs.putBoolean("firstrun", false);
-                    updateAllDatabases();
```

Comment:
I would move everything but logoutListener to the new callable to preserve the original ordering, juuuust in case that actually matters

Reply:
Done:)

Reply:
There has been a misunderstanding.

Only logoutListener should be in the subscribe/mainThread

Reply:
>preserve the original ordering

This has not been done, if we move code we shouldn't shuffle the lines up

Reply:
Also please add some logging to the error emission of the Completable subscription, this is what was making this error difficult to detect

Reply:
>This has not been done, if we move code we shouldn't shuffle the lines up
@macgills should I do this  in order?
```
 sessionManager.logout()
            .andThen(Completable.fromAction(() ->{
                Timber.d("All accounts have been removed");
                clearImageCache();
                //TODO: fix preference manager
                defaultPrefs.clearAll();
                defaultPrefs.putBoolean("firstrun", false);
                updateAllDatabases();

                }
            ))
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(() -> {
                    logoutListener.onLogoutComplete();
                });
```

Reply:
![image](https://user-images.githubusercontent.com/3358282/79210890-63c23c00-7e3d-11ea-9717-09b05ce00bbc.png)
Yes, as it originally appeared